drivers/sensor/ens210/ens210.c

```diff
@@ -46,6 +46,39 @@ static int ens210_sample_fetch(struct device *dev, enum sensor_channel chan)
 	u32_t temp_valid, humidity_valid;
 #endif /* CONFIG_ENS210_CRC_CHECK */
 
+#if defined(CONFIG_ENS210_TEMPERATURE_SINGLE) \
+	|| defined(CONFIG_ENS210_HUMIDITY_SINGLE)
+
+	const struct ens210_sens_start sense_start = {
+		.t_start = ENS210_T_START,
+		.h_start = ENS210_H_START
+	};
+
+	/* Start measuring */
+	ret = i2c_reg_write_byte(drv_data->i2c,
+			DT_INST_0_AMS_ENS210_BASE_ADDRESS,
+			ENS210_REG_SENS_START, *(u8_t *)&sense_start);
+	if (ret < 0) {
+		LOG_ERR("Failed to set SENS_START to 0x%x",
+				*(u8_t *)&sense_start);
+		return -EIO;
+	}
+
+	u8_t buf;
+
+	do {
+		k_sleep(K_MSEC(2));
+		ret =  i2c_reg_read_byte(drv_data->i2c,
+				DT_INST_0_AMS_ENS210_BASE_ADDRESS,
+				ENS210_REG_SENS_START, &buf);
+
+		if (ret < 0) {
+			LOG_ERR("Failed to read SENS_STAT");
+		}
+	} while (buf);
```

Comment:
Check only the bits where we measure in single-shot mode

Reply:
You're right, in case one sensor is in continuous mode, and the other is in single-shot mode this leads to an infinte loop. I'll change that..